main.go

```diff
@@ -67,14 +71,38 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func runPrometheusServer(metricsAddr string) {
+	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
+	// we need to unregister a couple of collectors, otherwise the metrics gathering will fail
+	// due to duplicate metrics collection
+	metrics.Registry.Unregister(prometheus.NewGoCollector())
```

Comment:
I wonder where the duplicates come from.
Is it the DefaultGatherer below?

Reply:
yes

Reply:
ok, I will remove it and if we need anything from it in the future we can add it then.